src/DIRC_geo.cpp

```diff
@@ -162,10 +158,16 @@ static Ref_t createDetector(Detector& desc, xml_h e, SensitiveDetector sens)
 
   double   lens_position_x = lens_shift;
   double   lens_position_z = -0.5 * (bar_assm_length + lens_thickness);
-  Position lens_position(lens_position_x, 0, lens_position_z);
-  dirc_module.placeVolume(lens_layer1_vol, lens_position);
-  dirc_module.placeVolume(lens_layer2_vol, lens_position);
-  dirc_module.placeVolume(lens_layer3_vol, lens_position);
+
+  for(int y_index = 0; y_index < bar_repeat_y; y_index++)
+    {
+      double lens_position_y = y_index*lens_width - 0.5*(prism_width - lens_width);
+
+      Position lens_position(lens_position_x, lens_position_y, lens_position_z);
+      dirc_module.placeVolume(lens_layer1_vol, lens_position);
+      dirc_module.placeVolume(lens_layer2_vol, lens_position);
+      dirc_module.placeVolume(lens_layer3_vol, lens_position);
+    }
```

Comment:
This should be changed and done elsewhere:
- place the lens layers in a single volume that is just the lens,
- place the lens volume in the loop earlier in the file where the bars are placed:
```
  for (int y_index = 0; y_index < bar_repeat_y; y_index++) {
    double y = 0.5 * bar_assm_width - 0.5 * bar_width - (bar_width + bar_gap) * y_index;
    for (int z_index = 0; z_index < bar_repeat_z; z_index++) {
      double z = 0.5 * bar_assm_length - 0.5 * bar_length - (bar_length + glue_thickness) * z_index;
      dirc_module.placeVolume(glue_vol, Position(0, y, z - 0.5 * (bar_length + glue_thickness)));
      dirc_module.placeVolume(bar_vol, Position(0, y, z)).addPhysVolID("section", z_index).addPhysVolID("bar", y_index);
    }
    dirc_module.placeVolume(lens_vol, Position(0, y, z_lens));
  }
```

Reply:
Is the suggestion to place lens layers in an "Assembly" like this?

Assembly lens_vol("Lens");
lens_vol.placeVolume(lens_layer1_solid, Position(0,0,0));

Reply:
No, that's not the suggestion. You shouldn't need an assembly here. Use volume hierarchy.